examples/fourier_transform_playground.py

```diff
@@ -140,12 +139,15 @@ def moving_wave(
         thread.send((wave_id, frequency, angle, phase_shift, speed))
 
 
-wdg = moving_wave()
-
 # add the widget to the window and run it once
-viewer.window.add_dock_widget(wdg, area='bottom')
-wdg()
-
-napari.run()
+viewer.window.add_dock_widget(moving_wave, area='bottom')
+moving_wave()
+
+if __name__ == '__main__':
+    napari.run()
+else:
+    # for the gallery, sleep to make sure that the images are captured
+    from time import time
+    time.sleep(5)
```

Comment:
Shouldn't `napari.run()` be called in this branch too?

Reply:
No. This branch is for the gallery, which has its own event loop mechanism.